httpx/utils.py

```diff
@@ -241,3 +250,54 @@ def elapsed(self) -> timedelta:
         if self.end is None:
             return timedelta(seconds=perf_counter() - self.start)
         return timedelta(seconds=self.end - self.start)
+
+
+ASGI_PLACEHOLDER_FORMAT = {
+    "body": "<{length} bytes>",
+    "bytes": "<{length} bytes>",
+    "text": "<{length} chars>",
+}
+
+
+def asgi_message_with_placeholders(message: dict) -> dict:
+    """
+    Return an ASGI message, with any body-type content omitted and replaced
+    with a placeholder.
+    """
+    new_message = message.copy()
+    for attr in ASGI_PLACEHOLDER_FORMAT:
```

Comment:
I think we need to have a function that can take something with an `.items()` method and filter out headers like `Authorization` and `Proxy-Authorization` like `Headers.__repr__()`.  Then we can use it in both places.